pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -89,7 +89,11 @@
       {"name", Oid.NAME, Types.VARCHAR, "java.lang.String", Oid.NAME_ARRAY},
       {"bytea", Oid.BYTEA, Types.BINARY, "[B", Oid.BYTEA_ARRAY},
       {"bool", Oid.BOOL, Types.BIT, "java.lang.Boolean", Oid.BOOL_ARRAY},
-      {"bit", Oid.BIT, Types.BIT, "java.lang.Boolean", Oid.BIT_ARRAY},
+      // Commented out this line to fix issue #892, an error that occurs when calling ResultSet getObject on a bit field.
```

Comment:
This would be breaking behaviour

Reply:
I told that myself as well, but I did scan the code to find all the places where `bit` and actually `varbit` are used and there aren't that many. Imo, the association: `{"bit", Oid.BIT, Types.BIT, "java.lang.Boolean", Oid.BIT_ARRAY}` is incorrect to begin with. ResultSet getObject doesn't currently work on bit fields because the code tries to coerce the value to a Boolean.  The information about the type `varbit` is retrieved dynamically (using queries against pg_catalog.pg_type & pg_catalog.pg_namespace) and essentially this change makes the driver treat `bit` in the same way as `varbit`.

Can you think of other test cases that I can add to test more to make sure nothing breaks? I added one where I wanted to make sure I can use `bit` as parameter. I also wanted to make sure getString works against the `bit` field.